### ISSUE TITLE
Atom date formats must be in RFC 3339 format

### ISSUE THREAD

**Issue by chriswill** [dotnet/SyndicationFeedReaderWriter#10]
I see that the library is using the "r" DateTime formatting option to convert datetimes to strings.   This is causing validation errors on the the W3C feed validation service: [https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fwww.cloudscope.io%2Fblog%2Ffeed](url).

I implemented some date formatting using XmlConvert.ToString that appears to pass:
`await writer.Write(new SyndicationContent("updated", XmlConvert.ToString(firstPost.CreatedDateUTC, XmlDateTimeSerializationMode.Utc)));`

But I don't have any way to do this for IAtomEntry's.

The Atom RFC requires 3339 format: [https://tools.ietf.org/html/rfc4287#page-9](url)

Would you like a PR?

**Comment by drago-draganov** [comment]
Thanks for reporting this, @chriswill!
To workaround the issue, please extend the built-in AtomFormatter. It's purpose is to allow customization. It will apply to the entire feed processing (including WriteUpdated and WriteItem).

```cs
class MyAtomFormatter : AtomFormatter {
        public override string FormatValue<T>(T value) {
            if (type == typeof(DateTimeOffset))
            {
                // Ensure RFC 3339 date-time format
                DateTimeOffset dto = (DateTimeOffset)(object)value;
                return XmlConvert.ToString(dto, "yyyy-MM-ddTHH:mm:sszzzzzzz");
            }
            
            // Fallback 
            return base.FormatValue(value);
        }
}  
```
 
```cs
//
// Usage

var writer = new AtomWriter(xmlWriter, null, new MyAtomFormatter());
//
// Do writes as usual
// ...

```

**Comment by zhenlan** [comment]
@drago-draganov, do you see a change is needed here and if so do you like @chriswill to do it?

@chriswill, thanks for your contribution! In the future, it will be great if you can get positive confirmation in the discussion that the issue is up-for-grab before sending a PR so we all can be sure time is spent on the right thing.

**Comment by drago-draganov** [comment]
A fix is in the works.

**Comment by drago-draganov** [comment]
Fixed in [commit](https://github.com/dotnet/SyndicationFeedReaderWriter/pull/12/commits/209834fcaace06ea9c6b4da7a1d397d818ef32a0)

**Comment by drago-draganov** [comment]
Fixed in #12
